compiler/rustc_type_ir/src/const_kind.rs

```diff
@@ -204,3 +204,36 @@ impl<CTX> HashStable<CTX> for InferConst {
         }
     }
 }
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "nightly", derive(HashStable_NoContext, TyEncodable, TyDecodable))]
+pub enum BoundConstness {
```

Comment:
idk how I feel about `BoundConstness` being part of `const_kind`, maybe move it to `const_trait` module or sth?